halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/PersistentStorageController.java

```diff
@@ -82,6 +83,49 @@
     builder.setRevert(() -> halconfigParser.undoChanges());
     builder.setSave(() -> halconfigParser.saveConfig());
 
-    return DaemonTaskHandler.submitTask(builder::build, "Edit storage settings");
+    return DaemonTaskHandler.submitTask(builder::build, "Set current persistent store");
+  }
+
+  @RequestMapping(value = "/persistentStore/{persistentStoreName:.+}", method = RequestMethod.GET)
+  DaemonTask<Halconfig, PersistentStore> getPersistentStore(@PathVariable String deploymentName,
```

Comment:
It's used to generate the config for the other services -- so it would be nice to avoid the `enable/disable` flags in the halconfig, and then only have them show when the front50.yml is written using the corresponding profile factory. In fact, I'd recommend removing those flags from the halconfig since they'll likely only cause confusion (but I understand why you've added them).